### PR TITLE
Bump azure compute version to 2022-03-01

### PR DESCRIPTION
**Pull request by MartinForReal**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Bump azure compute version to 202-03-01
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
bump azure compute version to 2022-03-01
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by feiskyer** [comment]
Some e2e tests have failed, are they related to the api version upgrade?

And could you add a release note?

File: pkg/azureclients/diskclient/interface.go [hunk 1]
```diff
@@ -19,7 +19,7 @@ package diskclient
 import (
 	"context"
 
-	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2021-12-01/compute"
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2022-03-01/compute"
```
Comment (feiskyer) [inline pkg/azureclients/diskclient/interface.go:22]: The APIVersion in L29 is "2022-03-02", @andyzhangx which is the correct one we should use?

Comment (andyzhangx) [reply pkg/azureclients/diskclient/interface.go:22]: apiVersion for disk is correct, it's `2022-03-02`, the sdk path is just `2022-03-01`, that's not related.
https://learn.microsoft.com/en-us/azure/templates/microsoft.compute/2022-03-02/disks

**Review by feiskyer — APPROVED** [review]
/lgtm

**Comment by MartinForReal** [comment]
<html>
<body>


/test pull-cloud-provider-azure-e2e-ccm-capz
--


<br class="Apple-interchange-newline">
</body>
</html>

**Comment by andyzhangx** [comment]
could you fix the snapshot api version? thanks  @MartinForReal 

```
                  Message: "create snapshot error: Retriable: false, RetryAfter: 0s, HTTPStatusCode: 400, RawError: {\"error\":{\"code\":\"NoRegisteredProviderFound\",\"message\":\"No registered resource provider found for location 'eastus2' and API version '2022-03-01' for type 'snapshots'. The supported api-versions are '2016-04-30-preview, 2017-03-30, 2018-04-01, 2018-06-01, 2018-09-30, 2019-03-01, 2019-07-01, 2019-11-01, 2020-05-01, 2020-06-30, 2020-09-30, 2020-12-01, 2021-04-01, 2021-08-01, 2021-12-01, 2022-03-02, 2022-07-02'. The supported locations are 'southeastasia, eastus2, centralus, westeurope, eastus, northcentralus, southcentralus, westus, northeurope, eastasia, brazilsouth, westus2, westcentralus, ukwest, uksouth, japaneast, japanwest, canadacentral, canadaeast, centralindia, southindia, australiaeast, australiasoutheast, koreacentral, koreasouth, westindia, francecentral, southafricanorth, uaenorth, australiacentral, switzerlandnorth, germanywestcentral, norwayeast, jioindiawest, westus3, swedencentral, qatarcentral, eastus2euap, centraluseuap'.\"}}",
```

**Comment by andyzhangx** [comment]
> could you fix the snapshot api version? thanks @MartinForReal
> 
> ```
>                   Message: "create snapshot error: Retriable: false, RetryAfter: 0s, HTTPStatusCode: 400, RawError: {\"error\":{\"code\":\"NoRegisteredProviderFound\",\"message\":\"No registered resource provider found for location 'eastus2' and API version '2022-03-01' for type 'snapshots'. The supported api-versions are '2016-04-30-preview, 2017-03-30, 2018-04-01, 2018-06-01, 2018-09-30, 2019-03-01, 2019-07-01, 2019-11-01, 2020-05-01, 2020-06-30, 2020-09-30, 2020-12-01, 2021-04-01, 2021-08-01, 2021-12-01, 2022-03-02, 2022-07-02'. The supported locations are 'southeastasia, eastus2, centralus, westeurope, eastus, northcentralus, southcentralus, westus, northeurope, eastasia, brazilsouth, westus2, westcentralus, ukwest, uksouth, japaneast, japanwest, canadacentral, canadaeast, centralindia, southindia, australiaeast, australiasoutheast, koreacentral, koreasouth, westindia, francecentral, southafricanorth, uaenorth, australiacentral, switzerlandnorth, germanywestcentral, norwayeast, jioindiawest, westus3, swedencentral, qatarcentral, eastus2euap, centraluseuap'.\"}}",
> ```

@umagnus could you help to make this change? it's a good way to get familiar with this snapshot client in azure lib.

**Comment by MartinForReal** [comment]
https://github.com/kubernetes-sigs/cloud-provider-azure/pull/2647 PTAL

**Pull request merged**